meson.build

```
@@ -381,11 +381,19 @@ luaold_interpreters = [
 
 lua_pc_name = ''
 
-if sys_windows or get_option('lua-interpreter') == 'lua'
-  if sys_windows and get_option('lua-interpreter') != 'lua'
-    warning('There is no "' + get_option('lua-interpreter')
-            + '" binding option for Windows, backing up to "lua"')
+if get_option('lua-interpreter') == 'luajit'
+  if sys_osx == true
```

If we are touching this, why not letting it more python-like:
```suggestion
  if sys_osx
```